include/seqan3/contrib/stream/bgzf_istream.hpp

```diff
@@ -0,0 +1,635 @@
+// zipstream Library License:
+// --------------------------
+//
+// The zlib/libpng License Copyright (c) 2003 Jonathan de Halleux.
+//
+// This software is provided 'as-is', without any express or implied warranty. In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose, including commercial applications, and to alter it and redistribute it freely, subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented; you must not claim that you wrote the original software. If you use this software in a product, an acknowledgment in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such, and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution
+//
+//
+// Author: Jonathan de Halleux, dehalleux@pelikhan.com, 2003   (original zlib stream)
+// Author: David Weese, dave.weese@gmail.com, 2014             (extension to parallel block-wise compression in bgzf format)
+// Author: Rene Rahn, rene.rahn AT fu-berlin.de, 2019          (adaption to SeqAn3)
+
+#pragma once
+
+// TODO(rrahn): replace me with constructor argument
+#ifndef SEQAN_BGZF_NUM_THREADS
+    #define SEQAN_BGZF_NUM_THREADS 16
+#endif
```

Comment:
Can you replace this with an `inline static const` global variable that is initialised with `std::thread::hardware_concurrency()`? I think that would make a lot of sense now that we have that feature or not?

Reply:
Well, I left everything as it was in SeqAn2. I first thought about extending the constructors of the stream to pass the thread number from outside, and actually even further to provide an executor interface later. But that would make the interface for the alignment file also more complicated because you would have to specify the thread num/executor there as well. Until we have a proper concept for supporting parallelism throughout all our files I think the solution with the `inline static` variable is very good. But I would not make it const, so people can override it in their program by assigning another value to it.

Reply:
> I first thought about extending the constructors of the stream to pass the thread number from outside, 

It's already a constructor arg of the streambuffer! One would only need to pass it through, but it's not important for the release.

> But that would make the interface for the alignment file also more complicated because you would have to specify the thread num/executor there as well.

Yes, I already thought of it. My plan was to simply add a `threads` member variable to the options of the files.  But this is also not relevant for the release and I am sure you have more ideas for this :wink: 

> Until we have a proper concept for supporting parallelism throughout all our files I think the solution with the inline static variable is very good. But I would not make it const, so people can override it in their program by assigning another value to it.

It's a good intermediate solution! We should definitely change it in the long run, though.